cpp/include/cudf/rolling.hpp

```diff
@@ -614,5 +627,52 @@ std::unique_ptr<column> rolling_window(
   rmm::cuda_stream_view stream      = cudf::get_default_stream(),
   rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Computes preceding and following window bounds from an input column.
+ *
+ * This function computes preceding and following window bounds, suitable for passing to
+ * `rolling_window` from a pair of scalars providing the window `length` and the `offset` of the
+ * window. Consider an element `pi` at index `i` of `input`. The window defined by `length` and
+ * `offset` is the ordered set of indices `Sj[i] := {j : input[j] \in [[pi + offset, ..., pi +
+ * offset + length]]}`. The returned `preceding_window` column is an INT32 column whose ith entry is
+ * `inf Sj[i] + 1` (the `+ 1` accounts for the usage in `rolling_window`), the returned
+ * `following_window` column contains as its ith entry `sup Sj[i]`.
+ *
+ * The endpoints of each window can be controlled via the `window_type` parameter:
+ * - If the window is `LEFT_CLOSED`, the condition is `input[j] \in [pi + offset, pi + offset +
+ * length)`
+ * - If the window is `RIGHT_CLOSED`, the condition is `input[j] \in (pi + offset, pi + offset +
+ * length]`
+ * - If the window is `CLOSED`, the condition is `input[j] \in [pi + offset, pi + offset + length]`
+ * - If the window is `OPEN`, the condition is `input[j] \in (pi + offset, pi + offset + length)`
+ *
+ * The input column can either be a timestamp type in which case length and offset must be durations
+ * of the same resolution as the timestamp; or a signed index type in which case length and offset
+ * must have the same index type.
+ *
+ * The input column must be sorted in ascending order (not checked), behaviour is undefined if this
+ * is not upheld.
+ *
```

Comment:
Include `@throw` tags here to specify what conditions this will API will throw an exception.

Reply:
Done.